From 0b2f550f6b539d759d200934a064a629807f77b3 Mon Sep 17 00:00:00 2001
From: Frank Ramsay <framsay@redhat.com>
Date: Thu, 9 Oct 2014 19:37:19 -0400
Subject: [fs] exec: kill the unnecessary mm->def_flags setting in load_elf_binary()

Message-id: <1412883441-28028-3-git-send-email-framsay@redhat.com>
Patchwork-id: 97305
O-Subject: [RHEL7.1 PATCH 2/4] exec: kill the unnecessary mm->def_flags setting in load_elf_binary()
Bugzilla: 1149786
RH-Acked-by: Oleg Nesterov <oleg@redhat.com>
RH-Acked-by: Larry Woodman <lwoodman@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

Bugzilla: 1149786
Upstream Status: upstream commit: ab0e113f6bee71a3933755d2c9ae41fcee631800
Build Info: 8080538
Tested: Tested on SGI UV system.

Commit: ab0e113f6bee71a3933755d2c9ae41fcee631800
Author: Alex Thorlton <athorlton@sgi.com>
Date: Mon, 7 Apr 2014 15:37:12 -0700

load_elf_binary() sets current->mm->def_flags = def_flags
and def_flags is always zero.  Not only this looks strange, this is
unnecessary because mm_init() has already set ->def_flags = 0.

 Signed-off-by: Alex Thorlton <athorlton@sgi.com>
 Suggested-by: Oleg Nesterov <oleg@redhat.com>
 Cc: Gerald Schaefer <gerald.schaefer@de.ibm.com>
 Cc: Martin Schwidefsky <schwidefsky@de.ibm.com>
 Cc: Heiko Carstens <heiko.carstens@de.ibm.com>
 Cc: Christian Borntraeger <borntraeger@de.ibm.com>
 Cc: Paolo Bonzini <pbonzini@redhat.com>
 Cc: "Kirill A. Shutemov" <kirill.shutemov@linux.intel.com>
 Cc: Mel Gorman <mgorman@suse.de>
 Acked-by: Rik van Riel <riel@redhat.com>
 Cc: Ingo Molnar <mingo@kernel.org>
 Cc: Peter Zijlstra <peterz@infradead.org>
 Cc: Andrea Arcangeli <aarcange@redhat.com>
 Cc: Oleg Nesterov <oleg@redhat.com>
 Cc: "Eric W. Biederman" <ebiederm@xmission.com>
 Cc: Alexander Viro <viro@zeniv.linux.org.uk>
 Cc: Johannes Weiner <hannes@cmpxchg.org>
 Cc: David Rientjes <rientjes@google.com>
 Cc: Paolo Bonzini <pbonzini@redhat.com>
 Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
 Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/binfmt_elf.c b/fs/binfmt_elf.c
index 3aac8e9..358e9af 100644
--- a/fs/binfmt_elf.c
+++ b/fs/binfmt_elf.c
@@ -582,7 +582,6 @@ static int load_elf_binary(struct linux_binprm *bprm)
  unsigned long start_code, end_code, start_data, end_data;
  unsigned long reloc_func_desc __maybe_unused = 0;
  int executable_stack = EXSTACK_DEFAULT;
- unsigned long def_flags = 0;
  struct pt_regs *regs = current_pt_regs();
  struct {
   struct elfhdr elf_ex;
@@ -722,9 +721,6 @@ static int load_elf_binary(struct linux_binprm *bprm)
  if (retval)
   goto out_free_dentry;
 
- /* OK, This is the point of no return */
- current->mm->def_flags = def_flags;
-
  /* Do this immediately, since STACK_TOP as used in setup_arg_pages
     may depend on the personality.  */
  SET_PERSONALITY(loc->elf_ex);
-- 
1.7.1